File: Model/src/main/java/com/promotick/lafabril/model/reporte/ReporteMundialTrivia.java
package com.promotick.lafabril.model.reporte;

import com.promotick.apiclient.utils.file.excel.annotations.ExcelClass;
import com.promotick.apiclient.utils.file.excel.annotations.ExcelColumn;
import lombok.Data;

import java.io.Serializable;

@Data
@ExcelClass(filename = "Reporte-mundial-trivia", sheetName = "Reporte Mundial Trivia")
public class ReporteMundialTrivia implements Serializable{

	private static final long serialVersionUID = 2068885050024304718L;

	private Integer idParticipante;

	@ExcelColumn("Catalogo")
	private String nombreCatalogo;

	@ExcelColumn("Usuario")
	private String usuarioParticipante;

	@ExcelColumn("Nombre")
	private String nombreParticipante;

	@ExcelColumn("Apellido Paterno")
	private String appaterno;

	@ExcelColumn("Apellido Materno")
	private String apmaterno;

	@ExcelColumn("Broker")
	private String broker;

	@ExcelColumn("Ciudad")
	private String ciudad;

	@ExcelColumn("Regional")
	private String regional;

	@ExcelColumn("Estado")
	private String statusParticipante;

	@ExcelColumn("Fecha Respuesta")
	private String fechaRespuesta;

	@ExcelColumn("Hora Respuesta")
	private String horaRespuesta;

	@ExcelColumn("Trivia")
	private String nombreTrivia;

	@ExcelColumn("Completado")
	private String completado;

	@ExcelColumn("Pregunta 1")
	private String respuesta1;

	@ExcelColumn("Pregunta 2")
	private String respuesta2;

	@ExcelColumn("Pregunta 3")
	private String respuesta3;

	@ExcelColumn("Pregunta 4")
	private String respuesta4;

	@ExcelColumn("Pregunta 5")
	private String respuesta5;

	@ExcelColumn("Ganador")
	private String ganador;
}

File: Resources/WebContent/public/admin/js/views/reporte-mundial-trivia.js
function ReporteMundialTrivia(uri) {
    this.uri = uri;
    this.rangoDatePicker = $('#rangoDatePicker');
    this.grid = $('#grid');
    this.btnFiltrar = $('#btnFiltrar');
    this.fechaInicio = $('#fechaInicio');
    this.fechaFin = $('#fechaFin');
    this.btnDescargar = $('#btnDescargar');
    this.inptBuscar = $('#buscar');
    this.btnBuscar = $('#btnBuscar');

    this.slcTrivia = $('#slcTrivia');
}

ReporteMundialTrivia.prototype.init = function() {
    this.handle();
    this.listar();
};

ReporteMundialTrivia.prototype.handle = function() {
    var obj = this;

    obj.rangoDatePicker.datepicker({
        endDate: "0m"
    });

    obj.btnFiltrar.on('click', function(e){
        e.preventDefault();
        obj.gridReload();
    });

    obj.btnBuscar.on('click', function(e){
        e.preventDefault();
        obj.gridReload();
    });

    obj.slcTrivia.on('change', function() {
        obj.gridReload();
    });

    obj.btnDescargar.on('click', function(e) {
        e.preventDefault();
        window.location.href = $(this).attr('data-descarga');
    })

};

ReporteMundialTrivia.prototype.listar = function(){
    var obj = this;
    obj.grid.DataTable({
        "bProcessing" : true,
        "bFilter" : false,
        "bLengthChange": false,
        "bPaginate": true,
        "bInfo" : true,
        "serverSide" : true,
        "bSort": false,
        "ajax": {
            "url" : obj.uri + 'reportes/mundial-trivia/listar',
            "type": "POST",
            "data" : function(d) {
                d.nroTrivia = $('#slcTrivia').find(":selected").val();
            }
        },
        "columns" : [
            {"data" : "nombreCatalogo", "orderable": false},
            {"data" : "usuarioParticipante", "orderable": false},
            {"data" : "nombreParticipante", "orderable": false},
            {"data" : "appaterno", "orderable": false},
            {"data" : "apmaterno", "orderable": false},
            {"data" : "broker", "orderable": false},
            {"data" : "ciudad", "orderable": false},
            {"data" : "regional", "orderable": false},
            {"data" : "statusParticipante", "orderable": false},
            {"data" : "fechaRespuesta", "orderable": false, "render": this._formatNull},
            {"data" : "horaRespuesta", "orderable": false, "render": this._formatNull},
            {"data" : "nombreTrivia", "orderable": false, "render": this._formatNull},
            {"data" : "completado", "orderable": false, "render": this._formatNull},
            {"data" : "respuesta1", "orderable": false, "render": this._formatNR},
            {"data" : "respuesta2", "orderable": false, "render": this._formatNR},
            {"data" : "respuesta3", "orderable": false, "render": this._formatNR},
            {"data" : "respuesta4", "orderable": false, "render": this._formatNR},
            {"data" : "respuesta5", "orderable": false, "render": this._formatNR},
            {"data" : "ganador", "orderable": false, "render": this._formatNull},


        ],
        "lengthMenu": [[10, 15, 20], [10, 15, 20]],
        "initComplete":function(settings,json){

        },
        "fnDrawCallback": function(){

        }
    });
};

ReporteMundialTrivia.prototype.gridReload = function(){
    this.grid.DataTable().ajax.reload();
};

ReporteMundialTrivia.prototype._formatArchivo = function (data, type, full) {
    return '<a href="' + data + '" target="_blank" class="btn btn-sm btn-success"><i class="fa fa-download" aria-hidden="true"></i> Descargar</a>';
};

ReporteMundialTrivia.prototype._formatNull = function(data, type, full) {
    return data != null ? data : '-';
};

ReporteMundialTrivia.prototype._formatNR = function(data, type, full) {
    return data != null ? data : '-';
};